tests/ios/swiftgen/swiftgen.bzl

```diff
@@ -0,0 +1,187 @@
+    )
+
+    maybe(
+        http_archive,
+        name = "Yams",
+        sha256 = "b31b6df500d6191368c93f605690ca9857fff7c6fd1c8897e9765fb624535c63",
+        strip_prefix = "Yams-5.0.0",
+        url = "https://github.com/jpsim/Yams/archive/refs/tags/5.0.0.tar.gz",
+    )
+
+    maybe(
+        http_archive,
+        name = "Commander",
+        build_file = "//tests/ios/swiftgen:Commander.BUILD",
+        sha256 = "4243b0227e51b8ea60345eac3ec4a3ff4385435e86011f2b60273e16806af9a8",
+        strip_prefix = "Commander-0.9.1",
+        url = "https://github.com/kylef/Commander/archive/0.9.1.tar.gz",
+    )
+
+    maybe(
+        http_archive,
+        name = "SwiftGen",
+        build_file = "//tests/ios/swiftgen:SwiftGen.BUILD",
+        sha256 = "fa9377ebea5c0bea55d67671eb2f20491a49f3a3d90b086d75743bffc99507f0",
+        strip_prefix = "SwiftGen-6.4.0",
+        url = "https://github.com/SwiftGen/SwiftGen/archive/6.4.0.tar.gz",
+    )
+
+def _swiftgen_impl(ctx):
+    srcs = ctx.files.srcs
+
+    inputs = []
+    inputs.extend(srcs)
+    inputs.append(ctx.file.template_file)
+    output = ctx.outputs.out
+
+    # Figure out the source type from the input files
+    src_type = "json"
+    for f in srcs:
+        if ".xcassets/" in f.path:
+            src_type = "xcassets"
+            break
+        if f.path.endswith(".strings") or f.path.endswith(".stringsdict"):
+            src_type = "strings"
+            break
+
+    args = ctx.actions.args()
+    args.add("run")
+    args.add(src_type)
+    args.add("--output", output)
+    args.add("--templatePath", ctx.file.template_file)
+
+    for (key, value) in ctx.attr.template_params.items():
+        args.add_joined(
+            "--param",
+            (key, value),
+            join_with = "=",
+        )
+
+    valid_src_types = ["xcassets", "strings", "json"]
+
+    # Figure out the last argument that will be passed to the swiftgen command
+    if src_type == "xcassets":
+        xcassets_dirs = group_files_by_directory(
+            srcs,
+            ["xcassets"],
+            attr = "srcs",
+        ).keys()
+        if len(xcassets_dirs) != 1:
+            fail("The xcassets_swift's srcs should contain exactly one" +
+                 " directory named *.xcassets.")
+        args.add(xcassets_dirs[0])
+
+    elif src_type == "strings":
+        for f in srcs:
+            if f.path.endswith(".strings"):
+                args.add(f)
+            if f.path.endswith(".stringsdict"):
+                args.add(f)
+
+    elif src_type not in valid_src_types:
+        fail("Failed to figure out the resource type from the input files, " +
+             "or support for it has not been implemented: {}".format(src_type))
+
+    ctx.actions.run(
+        inputs = inputs,
+        outputs = [output],
+        executable = ctx.executable._swiftgen,
+        arguments = [args],
+        mnemonic = "SwiftGen",
+    )
+
+swiftgen = rule(
```

Comment:
Should we add this rule to the repository? Its from https://github.com/line/rules_apple_line but maybe it would be good for users to have this rule as well without having to pull another repo for it? 

Could also see wanting to keep this repo just focused on Apple specific rules so I'm fine with either direction!

Reply:
If this is a test rule, I think it'd be good to ensure that we can depend on `rules_apple_line` stuff in general and everything is handled.

Because it is a `test` dependency - it can go into the `WORKSPACE` file: consumers of `rules_ios` shouldn't need to add `rules_apple_line` because it's in the `test` dir I believe. If you checkout my PR to verify proto support - this is what we did for `rules_proto_grpc`